mantle/kola/harness.go

```diff
@@ -614,7 +614,9 @@ func filterTests(tests map[string]*register.Test, patterns []string, pltfrm stri
 				continue
 			}
 		}
-
+		if t.Nvme != QEMUOptions.Nvme {
```

Comment:
Hmm, this looks odd in that spot. Let's circle back to that.

Reply:
Nor this.

Well... I see what you're trying to do. So e.g. if the user runs `kola run ... --qemu-nvme=true`, it doesn't run e.g. the `basic` test. But I don't think this is worth trying to cover. I.e. it's OK that technically, running the `basic.nvme` test is equivalent to the `basic` test with `--qemu-nvme=true`. I see those flags more useful in the `kola qemuexec`/`cosa run` context IMO. For testing, we should put emphasis on "de-matrixing" the combinations into individual tests.